pkg/controller/machineset/controller.go

```diff
@@ -68,7 +76,18 @@ func (c *MachineSetControllerImpl) Reconcile(machineSet *v1alpha1.MachineSet) er
 		return err
 	}
 
-	return c.syncReplicas(machineSet, filteredMachines)
+	syncErr := c.syncReplicas(machineSet, filteredMachines)
+
+	ms := machineSet.DeepCopy()
+	newStatus := c.calculateStatus(ms, filteredMachines)
+
+	// Always updates status as machines come up or die.
+	_, err = updateMachineSetStatus(c.clusterAPIClient.ClusterV1alpha1().MachineSets(machineSet.Namespace), machineSet, newStatus)
+	if err != nil {
+		return fmt.Errorf("failed to update machine set status, %v", err)
```

Comment:
recommend replacing the "," with a ":" as it signifies the error is the reason for why we "failed to update". Also it matches the general style of errors in this project.

Reply:
changed.